scientific/src/commonMain/kotlin/formatter/CommonFormatter.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ Copyright 2022 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.scientific.formatter
+
+import com.splendo.kaluga.scientific.ScientificValue
+import com.splendo.kaluga.scientific.unit.ScientificUnit
+import kotlin.math.roundToLong
+
+private typealias CustomFormatHandler = (Number) -> String
+sealed class CommonFormatter(builder: Builder) : Formatter {
```

Comment:
Not sure if it is better naming. We have Formatter which is in the `scientific` package so it is already scientific. 
You suggest renaming CommonFormatter to scientific which will be in fact `com.splendo.kaluga.scientific.formatter.ScientificFormatter` so it is kind of double scientific 🤓 in this case.

In fact, any scientific formatter will be scientific but only one will have it exquisitely in the name.

If you want explicitly mention that it is a `ScientificFormatter`, it should be `Formatter`. In this case, `CommonFormatter` will be renamed to `Common ScientificFormatter `

Reply:
Not saying it's mandatory to copy java conventions (which are also never 100% consistent) but I guess it gives a model https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html

then it would be `com.splendo.kaluga.scientific.format.(Common)ScientificValueFormatter`

we do also have some date/time formatting I think, I'd check what we do there too.

Reply:
The more scientific the better 🤓
> so it is kind of double scientific 🤓 in this case.

Reply:
At first look I didn't look at the package and I thought that if `base` would have some formatting classes with similar name it would be confusing, but I guess that's not the case

Reply:
Renamed to _CommonScientificValueFormatter_